src/diffusers/models/attention.py

```diff
@@ -315,7 +321,8 @@ def forward(
             ).chunk(6, dim=1)
             norm_hidden_states = self.norm1(hidden_states)
             norm_hidden_states = norm_hidden_states * (1 + scale_msa) + shift_msa
-            norm_hidden_states = norm_hidden_states.squeeze(1)
+            if self.squeeze_hidden_states:
```

Comment:
I don't think we want to add a config for this
can you explain why do we not want to squeeze here? what is the shape of `norm_hidden_states` here?

Reply:
Latte supports both text-to-video and text-to-image generation. When text-to-image is generated, the second dimension is 1, which squeeze drops, causing subsequent model inference errors.

Reply:
Here is the shape of `norm_hidden_states`:
![image](https://github.com/huggingface/diffusers/assets/38418898/ed3d6e6a-457a-49e4-93ce-8055b9c59ad2)

`norm_hidden_states.squeeze(1)` will cause the inference error when performing the temporal transformer block.

Reply:
I also tested the `pixartalpha` pipeline and I found that this line of code is redundant and we can remove it:
https://github.com/huggingface/diffusers/blob/867a2b0cf918503df256c8c5ff4dc6cc862dd257/src/diffusers/models/attention.py#L318

Because I print the shape of `norm_hidden_states`:

![image](https://github.com/huggingface/diffusers/assets/38418898/2b2cdfff-5057-4aba-bc72-99d9eb07997b)

The second dimension is larger than 1. Thus, `squeeze` doesn't compress the second dimension.

Reply:
can we do 
```python
if norm_hidden_states.shape[1] == 1:
    norm_hidden_states = norm_hidden_states.squeeze(1)
```

this block is used by quiet a lot of models, not sure which one this line is needed;

Reply:
Actually, Latte needs to retain the second dimension. Can we do:

```
norm_hidden_states = norm_hidden_states.squeeze(1)
if norm_hidden_states.ndim == 2: # for Latte
    norm_hidden_states = norm_hidden_states.unsqueeze(1)
```

When the attention calculation is performed, the input needs to be three dimensions. This should not cause other models to run incorrectly.

Reply:
can we adjust the shape in LatteTransformer2DModel so that you don't need to change code here?

Reply:
thanks for explaining this to me!
so I looked into a little bit, I think that's only used for pixart and was introduced in this PR https://github.com/huggingface/diffusers/pull/5642/files

if it does not actually do anything, the best solution is to delete it 

cc @sayakpaul here can you look into if this line can be removed, since you added it here in the pixart PR. @maxin-cn printed out shapes (see https://github.com/huggingface/diffusers/pull/8404#discussion_r1629163350) and we think that line does not do anything because that dimension is not going to be `1`

Reply:
I added it because otherwise, there were some weird broadcasting problems. I will look into it.